src/functions/TestDrive.ps1

```diff
@@ -71,7 +71,8 @@ function Clear-TestDrive ([String[]]$Exclude, [string]$TestDrivePath) {
 function New-RandomTempDirectory {
     do {
         $tempPath = Get-TempDirectory
-        $Path = [IO.Path]::Combine($tempPath, ([Guid]::NewGuid()));
+        $dirName = [IO.Path]::GetRandomFileName().Substring(0, 4)
```

Comment:
Perhaps 6 or 8.

Reply:
I chose 4 because that's what the `System.IO.Path.GetTempFileName()` static method uses. It returns a filename with four unique hex characters. But I'm not against six or eight characters, either.

Reply:
Which implementation of GetTempFileName() are you looking at? .NET does not do it. And .NET Framework does that neither, by looking at the code, they both use cryptographically safe randomization and a file check.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L277-L289

Reply:
That said, running this on my computer produces 1-10 conflicts. Which we would resolve by file check, so it is not likely to cause too many problems. 

`1..1000000  | % {  [IO.Path]::GetRandomFileName().Substring(0, 4) } | unique | Measure-Object`

Reply:
I was looking at `[System.IO.Path]::GetTempFileName()` as the basis for "four is enough":

Here's Windows PowerShell 5.1:

```powershell
$ 1..5 | % { [IO.Path]::GetTempFileName() } | split-Path -leaf
tmp7358.tmp
tmp7359.tmp
tmp735A.tmp
tmp735B.tmp
tmp735C.tmp
```

Notice the four unique hex digits in each filename.

Other versions of PowerShell/.NET behave the same.

PowerShell 7.2.6 on Windows

```powershell
$  1..5 | % { [IO.Path]::GetTempFileName() } | split-Path -leaf
tmp2643.tmp
tmp2644.tmp
tmp2645.tmp
tmp2646.tmp
tmp2647.tmp
```

Looks like PowerShell 7.2.3 on Linux uses random chars:

```powershell
> 1..10 | % { [IO.Path]::GetTempFileName() } | split-Path -leaf
tmpGLw1dM.tmp
tmp9BzMhL.tmp
tmpNTQ1qL.tmp
tmpENSaxM.tmp
tmp2dHGmL.tmp
```